pkg/controller/vitessshard/reconcile_topo.go

```diff
@@ -63,40 +63,22 @@ func (r *ReconcileVitessShard) reconcileTopology(ctx context.Context, vts *plane
 	defer ts.Close()
 	wr := wrangler.New(logutil.NewConsoleLogger(), ts.Server, nil)
 
-	// Get the shard record.
-	if shard, err := ts.GetShard(ctx, keyspaceName, vts.Spec.Name); err == nil {
-		vts.Status.HasMaster = k8s.ConditionStatus(shard.HasMaster())
-		if shard.MasterAlias != nil {
-			vts.Status.MasterAlias = topoproto.TabletAliasString(shard.MasterAlias)
-		}
-
-		// Is the shard in the serving partition for any cell or tablet type?
-		if servingCells, err := ts.GetShardServingCells(ctx, shard); err == nil {
-			vts.Status.Idle = k8s.ConditionStatus(len(servingCells) == 0)
+	if *vts.Spec.TopologyReconciliation.PruneShardCells {
```

Comment:
We need to pull out the parts that read topo and update `vts.Status` and still do that. This flag should only make us skip the mutation of topo.

Reply:
Done. PTAL.